flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/lineage/KafkaDatasetIdentifierProvider.java

```diff
@@ -0,0 +1,15 @@
+package org.apache.flink.connector.kafka.lineage;
+
+import java.util.Optional;
+
+/** Contains method which allows extracting topic identifier. */
+public interface KafkaDatasetIdentifierProvider {
```

Comment:
Make sure to tag all public API with `@PublicEvolving`. It needs to be clearly visible if a user is supposed to touch the class or not (the easiest way is to not use public unless needed).